editor/src/components/canvas/canvas-strategies/strategies/ancestor-metastrategy.tsx

```diff
@@ -0,0 +1,115 @@
+import { MetadataUtils } from '../../../../core/model/element-metadata-utils'
+import * as EP from '../../../../core/shared/element-path'
+import { CSSCursor } from '../../canvas-types'
+import { highlightElementsCommand } from '../../commands/highlight-element-command'
+import { setCursorCommand } from '../../commands/set-cursor-command'
+import { MetaCanvasStrategy } from '../canvas-strategies'
+import {
+  getTargetPathsFromInteractionTarget,
+  InteractionCanvasState,
+  targetPaths,
+} from '../canvas-strategy-types'
+
+export function ancestorMetaStrategy(
+  allOtherStrategies: Array<MetaCanvasStrategy>,
+  level: number,
+): MetaCanvasStrategy {
+  return (canvasState, interactionSession, customStrategyState) => {
+    // Don't apply during insertion
+    if (canvasState.interactionTarget.type === 'INSERTION_SUBJECTS') {
+      return []
+    }
+
+    const targets = getTargetPathsFromInteractionTarget(canvasState.interactionTarget)
+
+    if (targets.length !== 1) {
+      return []
+    }
+
+    const target = targets[0]
+
+    // Avoid children of the storyboard
+    if (EP.isEmptyPath(target) || EP.isStoryboardPath(target) || EP.isStoryboardChild(target)) {
+      // TODO Maybe avoid root elements?
+      return []
+    }
+
+    // Is the selected element an only child?
+    const siblings = MetadataUtils.getSiblings(canvasState.startingMetadata, target)
+    if (siblings.length > 1) {
+      return []
+    }
+
+    // Is the selected element a flow layout element?
+    const targetMetadata = MetadataUtils.findElementByElementPath(
+      canvasState.startingMetadata,
+      target,
+    )
+    const isStaticLayout = !(
+      MetadataUtils.isPositionAbsolute(targetMetadata) ||
+      MetadataUtils.isPositionRelative(targetMetadata) ||
+      MetadataUtils.isParentYogaLayoutedContainerForElementAndElementParticipatesInLayout(
+        targetMetadata,
+      )
+    )
+
+    if (!isStaticLayout) {
+      return []
+    }
+
+    // TODO Should we also check isParentZeroSized || isParentContiguous?
+
+    // Time to offer up available strategies for the parent
+    const parentPath = EP.parentPath(target)
+    const ancestorTargetPaths = targetPaths([parentPath])
+    const adjustedCanvasState: InteractionCanvasState = {
+      ...canvasState,
+      interactionTarget: ancestorTargetPaths,
+    }
+
+    // Avoid a cyclic dependency by explicitly passing the other metastrategies when creating the next layer's meta strategy
+    const nextAncestorResult = ancestorMetaStrategy(allOtherStrategies, level + 1)(
+      adjustedCanvasState,
+      interactionSession,
+      customStrategyState,
+    )
+
+    if (nextAncestorResult.length > 0) {
+      // A length of > 0 means that we should be bubbling up to the next ancestor
+      return nextAncestorResult
```

Comment:
can you move the call of the recursive ancestorMetaStrategy in this if branch to make it lazy?